pkg/specgenutil/specgen.go

```diff
@@ -679,6 +680,20 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *entities.ContainerCreateOptions
 		s.VolumesFrom = c.VolumesFrom
 	}
 
+	// Support relative paths
+	for i, v := range c.Volume {
+		vol := strings.Split(v, ":")
+		path, err := filepath.EvalSymlinks(vol[0])
+		if err != nil {
+			return err
+		}
+		vol[0], err = filepath.Abs(path)
+		if err != nil {
+			return err
+		}
+		c.Volume[i] = strings.Join(vol, ":")
+	}
```

Comment:
Why is this not in parseVolumes()?

AFAIK there is a reason it does not support relative paths, if I set `-v val:/path`, val will always be treated as named volume never a file path.
And relative file paths with `-v ./val:/path` are already supported, no?
@mheon PTAL

Reply:
I think the reason I did not put this in `parseVolumes` is because I tried to put it at the root of the error then it would actually be a change in c/common.

I will change this to only accept `./....` type mounts.

Reply:
Ok the issue we are trying to fix is podman --remote run -v ./foobar ... is totally broken right now.
I think we should support strings that match the following:

\.
.*/.*

Reply:
This has to be evaluated on the client side as well for podman --remote.

Reply:
this is the proper place to make sure this works on `--remote` I just need to modify the code to only match certain regex

Reply:
IMO we should only support relative paths when they start with a dot `.`.
I am also not sure if this is the right place parsing in different places is bad.

Reply:
What about 
podman run -v maindir/subdir/subdir:/testdir alpine ls -l /testdir

Reply:
@rhatdan WDTY about just supporting `./`?

Reply:
Looks like Docker rejects them all:

```
sh-5.1# docker run -v a/b:/mnt alpine ls /mnt
docker: Error response from daemon: create a/b: "a/b" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.
sh-5.1# docker run -v ./a/b:/mnt alpine ls /mnt
docker: Error response from daemon: create ./a/b: "./a/b" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
See 'docker run --help'.
sh-5.1# cd a/b/
sh-5.1# docker run -v .:/mnt alpine ls /mnt
docker: Error response from daemon: create .: volume name is too short, names should be at least two alphanumeric characters.
See 'docker run --help'.
```

Reply:
> What about podman run -v maindir/subdir/subdir:/testdir alpine ls -l /testdir

that works as long as you are in the parent of `maindir`

Reply:
So I guess I can support only fields that begin with a `.`

Reply:
Docker does not allow relative paths at all, our decision to allow them was pretty early on (before we had an API to worry about) and we decided to allow anything with a `./` prefix. That continues to seem sane to me.